src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -539,6 +688,16 @@ partial class BoundLambda : ILambdaExpression
         IBlockStatement ILambdaExpression.Body => this.Body;
 
         protected override OperationKind ExpressionKind => OperationKind.LambdaExpression;
+
+        public override void Accept(IOperationVisitor visitor)
+        {
+            visitor.VisitLambdaExpression(this);
+        }
+
+        public override TResult Accept<TResult>(IOperationVisitor<TResult> visitor)
+        {
+            return visitor.VisitLambdaExpression(this);
+        }
     }
 
     partial class BoundConversion : IConversionExpression, IMethodBindingExpression
```

Comment:
Are `BoundConversion` instances either `IConversionExpression` or `IMethodBindingExpression` but not both? If so, consider wrapping the `BoundConversion` instance in an object that implements just one of the two interfaces. That would remove the checks of `this.ConversionKind` including in the `Accept` methods, and it would also allow analyzers to reliably determine if the expression is one or the other using `as`: `var m = expr as IMethodBindingExpression;`.